pkg/defaults/defaults.go

```diff
@@ -182,23 +179,3 @@ func populateDefsConfig(dir string) (map[string]v1.OperatorSource, map[string]ol
 	}
 	return opsrcDefinitions, catsrcDefinitions, config, nil
 }
-
-// RemoveObsoleteOpsrc removes any existing default OperatorSource
-// in the cluster that has been switched into a default CatalogSource
-func RemoveObsoleteOpsrc(kubeClient client.Client) error {
```

Comment:
This was added when the Operatorhub was enhanced to accept both operatorsources and CatalogSources. After that change, when an operatorsource was switched to catalogsource, the `obsolete` operatorsources were still hanging around, [so it was necessary to `RemoveObsoleteOpsrc`](https://github.com/operator-framework/operator-marketplace/pull/308).

With that context, it makes sense for 4.7 to remove this since we don't have opsrc in 4.7, and therefore no opportunity to switch opsrc to catsrc (and therefore no necessity to `RemoveObsoleteOpsrc`).
However, if we're going to backport this all the way till 4.5, we might want to keep this around. Otherwise removing this will re-introduce [this](https://bugzilla.redhat.com/show_bug.cgi?id=1831818) bug.

Reply:
It would make sense to have this removed on a separate PR. I'll create one for it